### PR TITLE
OSDOCS-294: Included CSI details.

### PR DESCRIPTION
**Pull request by huffmanca**
Included the CSI Details from 3.11 into 4.0. The content has also been modularized and updated from PR 13273.

This is for OS 4.1.

@jsafrane, I incorporated the changes from PR 13273 in this PR. In addition, I updated both references of `quay.io/jsafrane/cinder-csi-plugin` to `k8scloudprovider/cinder-csi-plugin:v0.3.0`, whereas the previous PR had only updated one reference. Let me know if this was incorrect.

In addition, should we update all references of `driver` to `Operator` ?

### CODE REVIEW

**Comment by jsafrane** [comment]
>  I updated both references of quay.io/jsafrane/cinder-csi-plugin to k8scloudprovider/cinder-csi-plugin:v0.3.0, whereas the previous PR had only updated one reference.

This looks correct.

> In addition, should we update all references of driver to Operator ?

No, CSI driver is not an operator. csi-operator should not be mentioned anywhere in the docs.

**Comment by huffmanca** [comment]
@jsafrane, Thank you for the review!

@openshift/team-documentation , would someone be able to peer review this one?

File: modules/persistent-storage-csi-example-deployment.adoc [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+= Example CSI deployment
+
+Since {product-title} does not ship with any CSI driver installed, this 
+example shows how to deploy a community driver for OpenStack Cinder in 
+{product-title}.
+
+.Procedure
+
+. Create a new project where the CSI components will run and a new service 
+account that will run the components. Explicit node selector is used to
```
Comment (ahardin-rh) [inline modules/persistent-storage-csi-example-deployment.adoc]: account will run the components? 

An explicit node selector?

File: modules/persistent-storage-csi-external-controllers.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+* An external CSI attacher container translates `attach` and `detach` 
+calls from {product-title} to respective `ControllerPublish` and 
+`ControllerUnpublish` calls to the CSI driver.
+* An external CSI provisioner container that translates `provision` and 
+`delete` calls from {product-title} to respective `CreateVolume` and 
+`DeleteVolume` calls to the CSI driver.
+* A CSI driver container
+
+The CSI attacher and CSI provisioner containers talk to the CSI driver
```
Comment (ahardin-rh) [inline modules/persistent-storage-csi-external-controllers.adoc]: s/talk to/communicate with

File: modules/persistent-storage-csi-external-controllers.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+the CSI driver to use credentials to the storage backend. Run the CSI 
+controller pods on infrastructure nodes so the credentials are never leaked
+to user processes, even in the event of a catastrophic security breach 
+on a compute node.
+====
+
+[NOTE]
+====
+The external attacher must also run for CSI drivers that do not support 
+third-party attach/detach operations. The external attacher will not issue
```
Comment (ahardin-rh) [inline modules/persistent-storage-csi-external-controllers.adoc]: `attach` or `detach` operations

**Comment by ahardin-rh** [comment]
@huffmanca Just a few minor comments from me. Otherwise, looks good! :shipit:

File: modules/persistent-storage-csi-example-deployment.adoc (hunk 1 above)
Comment (huffmanca) [reply modules/persistent-storage-csi-example-deployment.adoc]: I've adjusted the first sentence to be:

    Create a new project where the CSI components will run, and then create a new service account to run these components.

I've implemented the second suggestion.

File: modules/persistent-storage-csi-external-controllers.adoc (hunk 2 above)
Comment (huffmanca) [reply modules/persistent-storage-csi-external-controllers.adoc]: Adjusted this sentence to use the recommendation.

File: modules/persistent-storage-csi-external-controllers.adoc (hunk 3 above)
Comment (huffmanca) [reply modules/persistent-storage-csi-external-controllers.adoc]: This sentence has been corrected.

**Pull request merged**